HopsanGenerator/src/generators/HopsanFMIGenerator.cpp

```diff
@@ -1310,10 +1305,47 @@ bool HopsanFMIGenerator::generateToFmu(QString savePath, ComponentSystem *pSyste
 
 
     //------------------------------------------------------------------//
-    // Generate model file
+    // Generate model file and export assets, replacing asset paths in model
     //------------------------------------------------------------------//
 
-    genOK = generateModelFile(pSystem, savePath);
+    QDir saveDir(savePath);
+    std::list<hopsan::HString> assets = pSystem->getModelAssets();
+    QMap<QString, QString> assetsMap;
+    if (!assets.empty()) {
+        printMessage("Exporting model assets");
+    }
+    for (const auto& asset : assets) {
+        QFileInfo assetInfo(asset.c_str());
+        QString absSourcePath = pSystem->findFilePath(asset).c_str();
+        QString targetPath;
+        if (assetInfo.isAbsolute()) {
+            // For absolute windows paths, replace \ with /
+            targetPath = absSourcePath.replace(R"(\)", "/");
+            // For absolute windows paths, replace :/ with /
+            targetPath = absSourcePath.replace(":/", "/");
+            // For Unix paths remove leading /
```

Comment:
Because it will be appended to the resource path and I do not want // in that path. Also the new path will be written to the model xml, as a relative path, relative to the resources directory (which will be set at runtime). If the leading / remains, the path will be treated as absolute and Hopsan will not search for the file.